libs/messenger/include/messenger/mailbox.hpp

```diff
@@ -0,0 +1,203 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/mutex.hpp"
+#include "messenger/message.hpp"
+
+#include "core/service_ids.hpp"
+#include "muddle/muddle_endpoint.hpp"
+#include "muddle/muddle_interface.hpp"
+#include "muddle/rpc/client.hpp"
+#include "muddle/rpc/server.hpp"
+#include "muddle/subscription.hpp"
+
+#include <deque>
+#include <unordered_map>
+
+namespace fetch {
+namespace messenger {
+
+class MailboxInterface
```

Comment:
Maybe not in the context of the PR but I would move to separate header file

Reply:
:+1: